src/main/java/org/embulk/output/mailchimp/MailChimpClient.java

```diff
@@ -110,162 +103,147 @@ public void handleErrors(List<ErrorResponse> errorResponses)
+                    continue taskCategories;
                 }
             }
-            return categories;
+            throw new ConfigException("Invalid group category name: '" + taskCategoryName + "'");
         }
+        return categories;
     }
 
+    private List<Interest> fetchInterests(MailChimpRetryable retryable,
+                                          String listId,
+                                          String categoryId)
+            throws JsonProcessingException
+    {
+        return fetch(retryable,
+                "/lists/" + listId + "/interest-categories/" + categoryId + "/interests",
+                "interests",
+                Interest[].class);
+    }
     /**
      * Extract merge fields from the list, find correct merge fields from API and put into the map to use
      * Reference: https://developer.mailchimp.com/documentation/mailchimp/reference/lists/merge-fields/#read-get_lists_list_id_merge_fields
-     *
-     * @param task the task
-     * @return the map
-     * @throws JsonProcessingException the json processing exception
      */
-    public Map<String, MergeField> extractMergeFieldsFromList(PluginTask task) throws JsonProcessingException
+    Map<String, MergeField> mergeFieldByTag(PluginTask task) throws JsonProcessingException
     {
-        try (MailChimpRetryable retryable = new MailChimpRetryable(task)) {
-            int count = 100;
-            int offset = 0;
-            int page = 1;
-            boolean hasMore = true;
-            List<MergeField> allMergeFields = new ArrayList<>();
-
-            while (hasMore) {
-                String path = MessageFormat.format("/lists/{0}/merge-fields?count={1}&offset={2}",
-                                                   task.getListId(),
-                                                   count,
-                                                   offset);
-
-                JsonNode response = jsonParser.parseJsonObject(retryable.get(path));
-                MergeFields mergeFields = mapper.treeToValue(response,
-                                                             MergeFields.class);
-
-                allMergeFields.addAll(mergeFields.getMergeFields());
-
-                if (hasMorePage(mergeFields.getTotalItems(), count, page)) {
-                    offset = count;
-                    page++;
-                }
-                else {
-                    hasMore = false;
-                }
-            }
+        return convertMergeFieldToMap(
+                fetch(task,
+                        "/lists/" + task.getListId() + "/merge-fields",
+                        "merge_fields",
+                        MergeField[].class));
+    }
 
-            return convertMergeFieldToMap(allMergeFields);
+    /**
+     * Like {@link MailChimpClient#fetch(MailChimpRetryable, String, String, Class)},
+     * with an automatically created MailchimpRetryable
+     */
+    private <T> List<T> fetch(PluginTask task,
```

Comment:
Because this method uses only once by `mergeFieldByTag` should we merge it?

Reply:
I prefer let it be this time, that method is meaningful on its own. Retryable is forwarded on the other overload is just because I don't want to alter the previous behavior (reuse same retryable for categories and interests).